### ISSUE TITLE
`bind:value` changes object even if I don't change it

### ISSUE THREAD

**Issue by frederikhors** [rob-balfre/svelte-select#312]
I createad a REPL for a big problem I'm having using `svelte-select`.

- Steps:

1. open the REPL: https://svelte.dev/repl/b965c63e347649b99a3a2e0d7e248ec9?version=3.42.4
1. open the console
1. you can clearly see there are **2** changes: `2x "Inner, team changed:" ▶ Object { player: Object { … } }`
1. after 4 seconds a new change is coming because of `setTimeout`

I think this is not the way to go because I expect `team` in `Inner.svelte` to not change because nothing changed.

In fact if you comment out `<SelectPlayer bind:value={customTeam.player}/>` in `Inner.svelte` **or simply remove the `bind:`** from `bind:value` it works like I need.

I'm using `bind:value` because I have many select in a form and I don't wanna write many

- `onSelect1Select()` and `onSelect1Clear()`
- `onSelect2Select()` and `onSelect2Clear()`
- `onSelect3Select()` and `onSelect3Clear()`
- `onSelect4Select()` and `onSelect4Clear()`
- and so on...

`bind:value` is amazing but this behaviour made my form crazy! It changes all the form many times and creates bugs!

**I expect to see a change only when I select a new item in list.**

Am I wrong?

**Comment by FluffyDiscord** [comment]
use  `<svelte:options immutable={true}/>` in the Inner.svelte

https://svelte.dev/repl/ac5683f0c3a940218584f1b5f7d81107?version=3.42.4

**Comment by frederikhors** [comment]
Unfortunately this is not a solution in my case. If I use this things go wrong in my `SelectPlayer` component: if I use simple svelte code in it it doesn't update prop like: `isWaiting`, `hasError`.

**Comment by rob-balfre** [comment]
@frederikhors hard to follow the example but I guess it's similar to your other issue. You're using `bind:value` and when `value` changes in one place it's firing off reactive events up and down all your components that are sharing bound values?

**Comment by frederikhors** [comment]
Yes, but why it changes **TWO** times?

The frist is from `undefined` to team, and the second, why?

![Uploading image.png…]()

**Comment by frederikhors** [comment]
I'm using `on:select` only now.

**Comment by EskelCz** [comment]
running into the same issue, would also prefer one-way `bind:value`
